editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```
@@ -97,17 +96,86 @@ export function reparentStrategyForParent(
     ? 'use-strict-bounds'
     : 'allow-missing-bounds'
 
-  return parentIsFlexLayout
-    ? {
-        strategy: 'REPARENT_TO_FLEX',
-        missingBoundsHandling: missingBoundsHandling,
-        isFallback: false,
-      }
-    : {
-        strategy: 'REPARENT_TO_ABSOLUTE',
-        missingBoundsHandling: missingBoundsHandling,
-        isFallback: convertToAbsolute,
-      }
+  const flowParentReparentType = flowParentAbsoluteOrStatic(targetMetadata, parent)
+  const reparentAsStatic =
+    !convertToAbsolute && (parentIsFlexLayout || flowParentReparentType === 'REPARENT_AS_STATIC')
+
+  if (reparentAsStatic) {
+    return {
+      strategy: 'REPARENT_AS_STATIC',
+      missingBoundsHandling: missingBoundsHandling,
+      isFallback: false,
+    }
+  } else {
+    return {
+      strategy: 'REPARENT_AS_ABSOLUTE',
+      missingBoundsHandling: missingBoundsHandling,
+      isFallback: convertToAbsolute,
+    }
+  }
+}
+
+function flowParentAbsoluteOrStatic(
+  metadata: ElementInstanceMetadataMap,
+  parent: ElementPath,
+): 'REPARENT_AS_ABSOLUTE' | 'REPARENT_AS_STATIC' {
+  const parentMetadata = MetadataUtils.findElementByElementPath(metadata, parent)
+  const children = MetadataUtils.getChildren(metadata, parent)
+
+  const storyboardRoot = EP.isStoryboardPath(parent)
+  if (storyboardRoot) {
+    // always reparent as absolute to the Storyboard
+    return 'REPARENT_AS_ABSOLUTE'
+  }
+
+  if (parentMetadata == null) {
+    // should this throw an error?
```

If there is the possibility of some race conditions that could lead to this, then we should either fail gracefully (and not reparent at all), or we should throw an error.

---

I changed it to throw an error. it should be an impossible condition to trigger it, let's see :)